File: tools/template/src/Employment.scala
package _PROJ_CLASS_.stage1

import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql._

import org.tresamigos.smv._

/**
 * ETL Module Example
 */

object EmploymentRaw extends SmvModule("ETL Example: Employment") with SmvOutput {

  override def requiresDS() = Seq(employment);
 
  override def run(i: runParams) = {
    val df = i(employment)

    df.groupBy("ST").agg(
      first("ST"),
      sum("EMP") as "EMP"
    )
  }
}

File: tools/templates/enterprise/src/main/python/stage1/employment.py
import smv
import pyspark.sql.functions as F

from stage1 import inputdata

__all__ = ['EmploymentByState']

class EmploymentByState(smv.SmvModule, smv.SmvOutput):
    """Python ETL Example: employ by state"""

    def requiresDS(self):
        return [inputdata.Employment]

    def run(self, i):
        df = i[inputdata.Employment]
        return df.groupBy(F.col("ST")).agg(F.sum(F.col("EMP")).alias("EMP"))
